solver/src/solver/http_solver/settlement.rs

```diff
@@ -201,12 +206,18 @@ fn match_settled_prices(
         .flat_map(|order| {
             iter::once(order.order.buy_token).chain(iter::once(order.order.sell_token))
         })
-        .chain(executed_amms.0.iter().flat_map(|amm| amm.order.tokens))
         .chain(
             executed_amms
-                .1
                 .iter()
-                .flat_map(|amm| amm.order.reserves.keys().copied().collect::<Vec<H160>>()),
```

Comment:
I know this was already here and isn't related to this PR but I'm wondering now if it is correct that the prices hashmap we return from this function contains all of the reserve keys? Doesn't this only need to contain the prices from limit orders because those are what the smart contract needs?

Reply:
Great point. I will check this out!

Reply:
So, given the function signature
```rust
fn match_settled_prices(
    executed_limit_orders: &[ExecutedLimitOrder],
    executed_amms: &[ExecutedAmm],
    solver_prices: HashMap<H160, Price>,
) -> Result<HashMap<H160, U256>> {
```

It would be possible to get the tokens from the Limit order however, I suspect we might run into issues where the execution plan routes though a token which is not included in the limit orders.

On the other hand, we shouldn't need to supply prices for these amm-only tokens, so I will create a separate issue for this and experiment with it in a following PR.

Reply:
Issue filed here: #892